src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -31,6 +31,12 @@ public class RestoreCommand
         private readonly Dictionary<RestoreTargetGraph, Dictionary<string, LibraryIncludeFlags>> _includeFlagGraphs
             = new Dictionary<RestoreTargetGraph, Dictionary<string, LibraryIncludeFlags>>();
 
+        public KeyValuePair<string, Guid> ParentId { get; }
```

Comment:
these can be private

Reply:
Just in case the parent want to know child's id, for now it can be private